File: test-services/scala-play/app/services/EchoService.scala
package services

import java.time.LocalDate
import java.util.UUID
import javax.inject._
import scala.concurrent._
import services.echo._
import models._

@Singleton
class EchoService @Inject()()(implicit ec: ExecutionContext) extends IEchoService {
  override def echoBodyString(body: String): Future[String] = Future { body }

  override def echoBodyModel(body: Message): Future[Message] = Future { body }

  override def echoBodyArray(body: List[String]): Future[List[String]] = Future { body }

  override def echoBodyMap(body: Map[String, String]): Future[Map[String, String]] = Future { body }

  override def echoQuery(
                          intQuery: Int,
                          longQuery: Long,
                          floatQuery: Float,
                          doubleQuery: Double,
                          decimalQuery: BigDecimal,
                          boolQuery: Boolean,
                          stringQuery: String,
                          stringOptQuery: Option[String],
                          stringDefaultedQuery: String,
                          stringArrayQuery: List[String],
                          uuidQuery: java.util.UUID,
                          dateQuery: java.time.LocalDate,
                          dateArrayQuery: List[java.time.LocalDate],
                          datetimeQuery: java.time.LocalDateTime,
                          enumQuery: Choice): Future[Parameters] = Future {
    Parameters(
      intField = intQuery,
      longField = longQuery,
      floatField = floatQuery,
      doubleField = doubleQuery,
      decimalField = decimalQuery,
      boolField = boolQuery,
      stringField = stringQuery,
      stringOptField = stringOptQuery,
      stringDefaultedField = stringDefaultedQuery,
      stringArrayField = stringArrayQuery,
      uuidField = uuidQuery,
      dateField = dateQuery,
      dateArrayField = dateArrayQuery,
      datetimeField = datetimeQuery,
      enumField = enumQuery
    )
  }

  override def echoHeader(
                           intHeader: Int,
                           longHeader: Long,
                           floatHeader: Float,
                           doubleHeader: Double,
                           decimalHeader: BigDecimal,
                           boolHeader: Boolean,
                           stringHeader: String,
                           stringOptHeader: Option[String],
                           stringDefaultedHeader: String,
                           stringArrayHeader: List[String],
                           uuidHeader: java.util.UUID,
                           dateHeader: java.time.LocalDate,
                           dateArrayHeader: List[java.time.LocalDate],
                           datetimeHeader: java.time.LocalDateTime,
                           enumHeader: Choice): Future[Parameters] = Future {
    Parameters(
      intField = intHeader,
      longField = longHeader,
      floatField = floatHeader,
      doubleField = doubleHeader,
      decimalField = decimalHeader,
      boolField = boolHeader,
      stringField = stringHeader,
      stringOptField = stringOptHeader,
      stringDefaultedField = stringDefaultedHeader,
      stringArrayField = stringArrayHeader,
      uuidField = uuidHeader,
      dateField = dateHeader,
      dateArrayField = dateArrayHeader,
      datetimeField = datetimeHeader,
      enumField = enumHeader
    )
  }

  override def echoUrlParams(
                              intUrl: Int,
                              longUrl: Long,
                              floatUrl: Float,
                              doubleUrl: Double,
                              decimalUrl: BigDecimal,
                              boolUrl: Boolean,
                              stringUrl: String,
                              uuidUrl: java.util.UUID,
                              dateUrl: java.time.LocalDate,
                              datetimeUrl: java.time.LocalDateTime,
                              enumUrl: Choice): Future[UrlParameters] = Future {
    UrlParameters(
      intField = intUrl,
      longField = longUrl,
      floatField = floatUrl,
      doubleField = doubleUrl,
      decimalField = decimalUrl,
      boolField = boolUrl,
      stringField = stringUrl,
      uuidField = uuidUrl,
      dateField = dateUrl,
      datetimeField = datetimeUrl,
      enumField = enumUrl
    )
  }

  override def echoEverything(
                               uuidHeader: java.util.UUID,
                               datetimeHeader: java.time.LocalDateTime,
                               body: Message,
                               dateUrl: java.time.LocalDate,
                               decimalUrl: BigDecimal,
                               floatQuery: Float,
                               boolQuery: Boolean): Future[EchoEverythingResponse] = Future {
    EchoEverythingResponse.Ok(
      Everything(
        bodyField = body,
        floatQuery = floatQuery,
        boolQuery = boolQuery,
        uuidHeader = uuidHeader,
        datetimeHeader = datetimeHeader,
        dateUrl = dateUrl,
        decimalUrl = decimalUrl
      )
    )
  }

  override def sameOperationName(): Future[SameOperationNameResponse] = Future {
    SameOperationNameResponse.Ok()
  }
}

File: test-services/typescript/template/src/echo_service.ts
import * as service from './spec/echo'
import * as models from './spec/models'

export let echoService = (): service.EchoService => {
    let echoBodyString = async (params: service.EchoBodyStringParams): Promise<string> => {
        return params.body
    }

    let echoBodyModel = async (params: service.EchoBodyModelParams): Promise<models.Message> => {
        return {int_field: params.body.int_field, string_field: params.body.string_field}
    }

    let echoBodyArray = async (params: service.EchoBodyArrayParams): Promise<string[]> => {
        return params.body
    }

    let echoBodyMap = async (params: service.EchoBodyMapParams): Promise<Record<string, string>> => {
        return params.body
    }

    let echoQuery = async (params: service.EchoQueryParams): Promise<models.Parameters> => {
        return {
            int_field: params.int_query,
            long_field: params.long_query,
            float_field: params.float_query,
            double_field: params.double_query,
            decimal_field: params.decimal_query,
            bool_field: params.bool_query,
            string_field: params.string_query,
            string_opt_field: params.string_opt_query,
            string_defaulted_field: params.string_defaulted_query,
            string_array_field: params.string_array_query,
            uuid_field: params.uuid_query,
            date_field: params.date_query,
            date_array_field: params.date_array_query,
            datetime_field: params.datetime_query,
            enum_field: params.enum_query,
        }
    }

    let echoHeader = async (params: service.EchoHeaderParams): Promise<models.Parameters> => {
        return {
            int_field: params['Int-Header'],
            long_field: params['Long-Header'],
            float_field: params['Float-Header'],
            double_field: params['Double-Header'],
            decimal_field: params['Decimal-Header'],
            bool_field: params['Bool-Header'],
            string_field: params['String-Header'],
            string_opt_field: params['String-Opt-Header'],
            string_defaulted_field: params['String-Defaulted-Header'],
            string_array_field: params['String-Array-Header'],
            uuid_field: params['Uuid-Header'],
            date_field: params['Date-Header'],
            date_array_field: params['Date-Array-Header'],
            datetime_field: params['Datetime-Header'],
            enum_field: params['Enum-Header'],
        }
    }

    let echoUrlParams = async (params: service.EchoUrlParamsParams): Promise<models.UrlParameters> => {
        return {
            int_field: params.int_url,
            long_field: params.long_url,
            float_field: params.float_url,
            double_field: params.double_url,
            decimal_field: params.decimal_url,
            bool_field: params.bool_url,
            string_field: params.string_url,
            uuid_field: params.uuid_url,
            date_field: params.date_url,
            datetime_field: params.datetime_url,
            enum_field: params.enum_url,
        }
    }

    let echoEverything = async (params: service.EchoEverythingParams): Promise<service.EchoEverythingResponse> => {
        return {
            status: "ok",
            data: {
                body_field: params.body,
                float_query: params.float_query,
                bool_query: params.bool_query,
                uuid_header: params['Uuid-Header'],
                datetime_header: params['Datetime-Header'],
                date_url: params.date_url,
                decimal_url: params.decimal_url,
            }
        }
    }

    let sameOperationName = async (): Promise<service.SameOperationNameResponse> => {
        return {status: 'ok'}
    }

    return {echoBodyString, echoBodyModel, echoBodyArray, echoBodyMap, echoQuery, echoHeader, echoUrlParams, echoEverything, sameOperationName}
}